client/lib/importer/actions.js

```diff
@@ -2,20 +2,50 @@
  * External dependencies
  */
 import Dispatcher from 'dispatcher';
+import noop from 'lodash/utility/noop';
 const wpcom = require( 'lib/wp' ).undocumented();
 
 /**
  * Internal dependencies
  */
-import { actionTypes } from './constants';
+import { actionTypes, appStates } from './constants';
 import { fromApi, toApi } from './common';
 
-export function cancelImport( siteId, importerId ) {
+const ID_GENERATOR_PREFIX = 'local-generated-id-';
+
+const cancelOrder = ( siteId, importerId ) => toApi( { importerId, importerState: appStates.CANCEL_PENDING, site: { ID: siteId } } );
+const apiStart = () => Dispatcher.handleViewAction( { type: actionTypes.API_REQUEST } );
+const apiSuccess = () => Dispatcher.handleViewAction( { type: actionTypes.API_SUCCESS } );
+const apiFailure = () => Dispatcher.handleViewAction( { type: actionTypes.API_FAILURE } );
+
+function apiUpdateImporter( importerStatus ) {
+	apiSuccess();
+
+	Dispatcher.handleViewAction( {
+		type: actionTypes.RECEIVE_IMPORT_STATUS,
+		importerStatus
+	} );
+}
+
+export function cancelImport( siteId, importerId, uploadAborter = noop ) {
+	uploadAborter();
+
 	Dispatcher.handleViewAction( {
 		type: actionTypes.CANCEL_IMPORT,
 		importerId,
 		siteId
 	} );
+
+	if ( importerId.includes( ID_GENERATOR_PREFIX ) ) {
```

Comment:
thanks @aduth - I remember your posts but can't remember how long ago it was. Did you have some test suite at the time or is there any way to know if this _could_ have changed since then or if Babel 6 would make a difference?